lib/basedriver/commands/timeout.js

```diff
@@ -4,12 +4,30 @@ import B from 'bluebird';
 import _ from 'lodash';
 import { util } from 'appium-support';
 import { errors } from '../../mjsonwp';
+import BaseDriver from "../driver";
 
 
 let commands = {}, helpers = {}, extensions = {};
 
 const MIN_TIMEOUT = 0;
 
+commands.timeoutsW3C = async function (timeoutsObj) {
+  if (timeoutsObj.protocol === BaseDriver.DRIVER_PROTOCOL.W3C) {
+    const {script, pageLoad, implicit} = timeoutsObj;
+    if (util.hasValue(script) || util.hasValue(pageLoad)) {
+      throw new errors.NotImplementedError('Not implemented yet for script and pageLoad.');
+    }
+
+    if (util.hasValue(implicit)) {
+      this.setImplicitWait(this.parseTimeoutArgument(implicit));
+    }
+  } else {
+    const {type, ms} = timeoutsObj;
+    this.timeouts(type, ms);
+  }
+};
+
+// TODO: remove and rename timeoutsW3C to timeouts
```

Comment:
TODO left?

Reply:
Yes, just for memos I'll work after my work 😅